### PR TITLE
 Issue #1468 Do not create config for existing cachemanager

### PR DESCRIPTION
**Pull request by kcleereman**
Issue #1468 Do not create config for existing cachemanager

### CODE REVIEW

File: 107/src/main/java/org/ehcache/jsr107/EhcacheCachingProvider.java [hunk 1]
```diff
@@ -113,9 +143,14 @@ public Eh107CacheManager getCacheManager(URI uri, Configuration config) {
    * @return a cache manager
    */
   public Eh107CacheManager getCacheManager(URI uri, Configuration config, Properties properties) {
+    return getCacheManager(new ConfigSupplier(uri, config), properties);
+  }
+
+  public Eh107CacheManager getCacheManager(ConfigSupplier configSupplier, Properties properties) {
```
Comment (ljacomet) [inline 107/src/main/java/org/ehcache/jsr107/EhcacheCachingProvider.java]: This should not be public.

File: 107/src/main/java/org/ehcache/jsr107/EhcacheCachingProvider.java [hunk 2]
```diff
@@ -52,6 +52,47 @@
 
   private final Map<ClassLoader, ConcurrentMap<URI, Eh107CacheManager>> cacheManagers = new WeakHashMap<ClassLoader, ConcurrentMap<URI, Eh107CacheManager>>();
 
+  public static class ConfigSupplier {
```
Comment (ljacomet) [inline 107/src/main/java/org/ehcache/jsr107/EhcacheCachingProvider.java]: this should not be public either. And by usual code style in the project, should be at the end of the file.

**Comment by ljacomet** [comment]
Looks good in principle - just visibility and code style remarks

File: 107/src/main/java/org/ehcache/jsr107/EhcacheCachingProvider.java (hunk 1 above)
Comment (kcleereman) [inline 107/src/main/java/org/ehcache/jsr107/EhcacheCachingProvider.java]: I've made the visibility changes, not sure why github hasn't noticed but oh well

File: 107/src/main/java/org/ehcache/jsr107/EhcacheCachingProvider.java [hunk 3]
```diff
@@ -50,6 +51,8 @@
 
   private static final URI URI_DEFAULT;
 
+  private static final AtomicInteger CONFIG_COUNT = new AtomicInteger(0);
```
Comment (ljacomet) [inline 107/src/main/java/org/ehcache/jsr107/EhcacheCachingProvider.java]: @kcleereman Your previous version was better - just make the inner class and added getter package private. Here this atomic is still public and just plain useless but for the test purpose.

**Comment by ljacomet** [comment]
Good to go once rebased

**Pull request merged**